src/ai/backend/common/types.py

```diff
@@ -1279,3 +1279,29 @@ class ModelServiceProfile:
     ),
     RuntimeVariant.CMD: ModelServiceProfile(name="Predefined Image Command"),
 }
+
+
+@dataclass
+class SimpleKernelRegistry(JSONSerializableMixin):
```

Comment:
Since this does not manage the states but is just a collection of serializable information about the kernels, how about renaming it like a more data-structure-like one? e.g., `KernelRecords`, `KernelStatusCollection`, ...

Reply:
I want to allow any iterable data type here since sequential data fetched from RPC response get deserialized into `tuple`.
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/qqKjbYtIoo1cw3D1CMWW/f8dee5ec-7e6b-4717-979e-c93767430d07.png)